### ISSUE TITLE
[red-knot] No error when using some `types` symbols without importing them

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ruff#15476]
Red Knot does not raise any errors when using some symbols from `types` without importing them. I haven't done an exhaustive investigation, but all of these should raise errors:
```py
a: Literal[2] =   # no error here
b: Any = 2  # no error here
c: Never = 3
d: Annotated[int, "…"] = 2
e: Union[int, str] = 2
```

**Comment by AlexWaygood** [comment]
I think this is a duplilcate of https://github.com/astral-sh/ruff/issues/14099 -- these symbols are imported in typeshed's `builtins.pyi` file, and we haven't yet implemented an understanding the re-export conventions for stub files. (No import in a stub file should be understood as a public re-export unless it uses the explicit `import foo as foo` or `from foo import bar as bar` syntax)
